rosidl_generator_py/resource/msg.py.template

```diff
@@ -0,0 +1,27 @@
+# Generated by rosidl_generator_py
+
+class @(spec.base_type.type):
+@[if len(spec.fields) > 0]@
+    __slots__ = ['@("', '".join([field.name for field in spec.fields]))']
+    def __init__(self, **kwargs):
+@[for field in spec.fields]@
+@[if field.default_value]@
+        self.@(field.name) = kwargs.get(
+            '@(field.name)', @value_to_py(field.type, field.default_value))
+@[else]@
+        self.@(field.name) = kwargs.get('@(field.name)')
+@[end if]@
+@[end for]@
+@[else]@
+    __slots__ = []
+@[end if]@
+
+    def __eq__(self, other):
+        return (type(self) == type(other)
+@[for field in spec.fields]@
+            and self.@(field.name) == other.@(field.name)
+@[end for]@
```

Comment:
I meant why is implementing a custom `__eq__` necessary?

Also I don't think we should discourage people from subclassing from generated classes. There are use cases where that makes sense. Also if we would discourage that it would need to be mentioned (and argued for) in the article since it is quite a restriction.